DevModeInitializer generates ClassNotFoundException for WildFly builtin servlets

Vaadin 14.2.0.beta1
WildFly 16.0.0.Final

Greetings!

After upgrading Vaadin from 14.1.23 to 14.2.0.beta1 we have exceptions at deployment time

```
Caused by: javax.servlet.ServletException: Servlet class name (io.undertow.servlet.handlers.DefaultServlet) can't be found!
	at com.vaadin.flow.server.startup.DevModeInitializer.process(DevModeInitializer.java:194)
	... 26 more
Caused by: java.lang.ClassNotFoundException: io.undertow.servlet.handlers.DefaultServlet from [Module "deployment.openicarStart.ear" from Service Module Loader]
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:255)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.vaadin.flow.server.startup.DevModeInitializer.isVaadinServletSubClass(DevModeInitializer.java:214)
	at com.vaadin.flow.server.startup.DevModeInitializer.process(DevModeInitializer.java:187)
	... 26 more
```

at the same point

`java.lang.ClassNotFoundException: org.apache.jasper.servlet.JspServlet`

WildFly application server has class-loading isolation in modules and Vaadin's behavior in my opinion is not entirely correct. Can this be fixed (by ignoring ClassNotFoundException)?


Workaround that helps us is append dependencies to jboss-deployment-structure:

```
<module name="io.undertow.servlet" export="true" />
<module name="io.undertow.jsp" export="true" />
```

Thanks!

--- a/flow-server/src/main/java/com/vaadin/flow/server/startup/DevModeInitializer.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/startup/DevModeInitializer.java
@@ -227,7 +227,7 @@ public class DevModeInitializer
         try {
             return VaadinServlet.class
                     .isAssignableFrom(Class.forName(className));
-        } catch (ClassNotFoundException exception) {
+        } catch (ClassNotFoundException exception) {// NOSONAR
             log().debug(String.format("Servlet class name (%s) can't be found!",
                     className));
             return false;

--- a/flow-server/src/main/java/com/vaadin/flow/server/startup/DevModeInitializer.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/startup/DevModeInitializer.java
@@ -192,8 +192,7 @@ public class DevModeInitializer
     // This attribute helps to avoid Dev Mode running twice.
     //
     // Addresses the issue https://github.com/vaadin/spring/issues/502
-    private static final String DEV_MODE_HANDLER_ALREADY_STARTED_ATTRIBUTE =
-            "dev-mode-handler-already-started-attribute";
+    private static final String DEV_MODE_HANDLER_ALREADY_STARTED_ATTRIBUTE = "dev-mode-handler-already-started-attribute";
 
     @Override
     public void process(Set<Class<?>> classes, ServletContext context)
@@ -203,18 +202,10 @@ public class DevModeInitializer
 
         ServletRegistration vaadinServletRegistration = null;
         for (ServletRegistration registration : registrations) {
-            try {
-                if (registration.getClassName() != null
-                        && isVaadinServletSubClass(
-                                registration.getClassName())) {
-                    vaadinServletRegistration = registration;
-                    break;
-                }
-            } catch (ClassNotFoundException e) {
-                throw new ServletException(
-                        String.format("Servlet class name (%s) can't be found!",
-                                registration.getClassName()),
-                        e);
+            if (registration.getClassName() != null
+                    && isVaadinServletSubClass(registration.getClassName())) {
+                vaadinServletRegistration = registration;
+                break;
             }
         }
 
@@ -232,9 +223,16 @@ public class DevModeInitializer
         setDevModeStarted(context);
     }
 
-    private boolean isVaadinServletSubClass(String className)
-            throws ClassNotFoundException {
-        return VaadinServlet.class.isAssignableFrom(Class.forName(className));
+    private boolean isVaadinServletSubClass(String className) {
+        try {
+            return VaadinServlet.class
+                    .isAssignableFrom(Class.forName(className));
+        } catch (ClassNotFoundException exception) {
+            log().debug(String.format("Servlet class name (%s) can't be found!",
+                    className));
+            return false;
+
+        }
     }
 
     private void setDevModeStarted(ServletContext context) {
@@ -395,11 +393,13 @@ public class DevModeInitializer
     /**
      * Shows whether {@link DevModeHandler} has been already started or not.
      *
-     * @param servletContext The servlet context, not <code>null</code>
-     * @return <code>true</code> if {@link DevModeHandler} has already been started,
-     *         <code>false</code> - otherwise
+     * @param servletContext
+     *            The servlet context, not <code>null</code>
+     * @return <code>true</code> if {@link DevModeHandler} has already been
+     *         started, <code>false</code> - otherwise
      */
-    public static boolean isDevModeAlreadyStarted(ServletContext servletContext) {
+    public static boolean isDevModeAlreadyStarted(
+            ServletContext servletContext) {
         assert servletContext != null;
         return servletContext.getAttribute(
                 DevModeInitializer.DEV_MODE_HANDLER_ALREADY_STARTED_ATTRIBUTE) != null;

